core/src/main/java/com/netflix/iceberg/BaseMetastoreTables.java

```diff
@@ -21,45 +21,42 @@
 import com.netflix.iceberg.exceptions.NoSuchTableException;
 import org.apache.hadoop.conf.Configuration;
 
-public abstract class BaseMetastoreTables {
+public abstract class BaseMetastoreTables implements Tables {
   private final Configuration conf;
 
   public BaseMetastoreTables(Configuration conf) {
     this.conf = conf;
   }
 
-  public abstract BaseMetastoreTableOperations newTableOps(Configuration conf, String database, String table);
+  public abstract BaseMetastoreTableOperations newTableOps(Configuration conf, String location);
```

Comment:
I'd like to use the separate db and table strings where possible. I think this should probably just add `load(String)` that calls `load(String, String)`. Maybe we should have an optional interface for the latter to expose it when the processing engine has already parsed the identifier so that we don't need to make a string and parse it again.